cli/tests/test_git_fetch.rs

```diff
@@ -100,25 +95,47 @@ fn clone_git_remote_into(
     test_env: &TestEnvironment,
     upstream: &str,
     fork: &str,
-) -> git2::Repository {
+) -> gix::Repository {
     let upstream_path = test_env.env_root().join(upstream);
     let fork_path = test_env.env_root().join(fork);
-    let fork_repo = git2::Repository::init(fork_path).unwrap();
+    let fork_repo = git::init(fork_path);
     {
-        let mut upstream_remote = fork_repo
-            .remote(upstream, upstream_path.to_str().unwrap())
+        let upstream_remote = fork_repo
+            .remote_at(upstream_path.to_str().unwrap())
+            .unwrap();
+        git::add_remote(&fork_repo, upstream, upstream_remote.clone());
+        upstream_remote
+            .with_refspecs(
+                Some(format!("+refs/heads/*:refs/remotes/{upstream}/*").as_bytes()),
+                gix::remote::Direction::Fetch,
+            )
+            .unwrap()
+            .connect(gix::remote::Direction::Fetch)
+            .unwrap()
+            .prepare_fetch(
+                gix::progress::Discard,
+                gix::remote::ref_map::Options::default(),
+            )
+            .unwrap()
+            .receive(gix::progress::Discard, &gix::interrupt::IS_INTERRUPTED)
```

Comment:
Since this isn't trivial, I would extract `testutils::git::fetch()` function. It might be also better to shell out `git`.

`testutils::git::clone()` might just work, but I haven't checked.